File: Android/melody-core/src/main/java/com/melody/core/schema/ViewAlignment.kt
package com.melody.core.schema

/**
 * Alignment for views and containers.
 * Mirrors iOS `ViewAlignment`.
 */
enum class ViewAlignment(val value: String) {
    Center("center"),
    Top("top"),
    Bottom("bottom"),
    Leading("leading"),
    Left("left"),
    Trailing("trailing"),
    Right("right"),
    TopLeading("topLeading"),
    TopLeft("topLeft"),
    TopTrailing("topTrailing"),
    TopRight("topRight"),
    BottomLeading("bottomLeading"),
    BottomLeft("bottomLeft"),
    BottomTrailing("bottomTrailing"),
    BottomRight("bottomRight");

    companion object {
        fun from(rawValue: String): ViewAlignment {
            return when (rawValue.lowercase()) {
                "center" -> Center
                "top" -> Top
                "bottom" -> Bottom
                "leading" -> Leading
                "left" -> Left
                "trailing" -> Trailing
                "right" -> Right
                "topleading" -> TopLeading
                "topleft" -> TopLeft
                "toptrailing" -> TopTrailing
                "topright" -> TopRight
                "bottomleading" -> BottomLeading
                "bottomleft" -> BottomLeft
                "bottomtrailing" -> BottomTrailing
                "bottomright" -> BottomRight
                else -> Leading
            }
        }
    }
}

File: Sources/Core/Schema/Style/ViewAlignment.swift
import Foundation

/// Named alignment position for content within a container.
public enum ViewAlignment: String, Codable, Sendable, Equatable {
    case center
    case top
    case bottom
    case leading
    case left
    case trailing
    case right
    case topLeading
    case topLeft
    case topTrailing
    case topRight
    case bottomLeading
    case bottomLeft
    case bottomTrailing
    case bottomRight

    public init(rawValue: String) {
        switch rawValue {
        case "center": self = .center
        case "top": self = .top
        case "bottom": self = .bottom
        case "leading": self = .leading
        case "left": self = .left
        case "trailing": self = .trailing
        case "right": self = .right
        case "topLeading": self = .topLeading
        case "topLeft": self = .topLeft
        case "topTrailing": self = .topTrailing
        case "topRight": self = .topRight
        case "bottomLeading": self = .bottomLeading
        case "bottomLeft": self = .bottomLeft
        case "bottomTrailing": self = .bottomTrailing
        case "bottomRight": self = .bottomRight
        default: self = .leading
        }
    }

    public init(from decoder: Decoder) throws {
        let value = try decoder.singleValueContainer().decode(String.self)
        self.init(rawValue: value)
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encode(rawValue)
    }
}
